pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1224,19 +1224,17 @@ func storePipelineSpecAndMergeMeta(ctx context.Context, pr *v1beta1.PipelineRun,
 	// Propagate refSource from remote resolution to PipelineRun Status
 	// This lives outside of the status.spec check to avoid the case where only the spec is available in the first reconcile and source comes in next reconcile.
 	cfg := config.FromContextOrDefaults(ctx)
-	if cfg.FeatureFlags.EnableProvenanceInStatus {
-		if pr.Status.Provenance == nil {
-			pr.Status.Provenance = &v1beta1.Provenance{}
-		}
-		// Store FeatureFlags in the Provenance.
-		pr.Status.Provenance.FeatureFlags = cfg.FeatureFlags
+	if pr.Status.Provenance == nil {
+		pr.Status.Provenance = &v1beta1.Provenance{}
+	}
+	// Store FeatureFlags in the Provenance.
+	pr.Status.Provenance.FeatureFlags = cfg.FeatureFlags
 
-		if meta != nil && meta.RefSource != nil && pr.Status.Provenance.RefSource == nil {
-			pr.Status.Provenance.RefSource = meta.RefSource
-		}
-		if meta != nil && meta.RefSource != nil && pr.Status.Provenance.ConfigSource == nil {
-			pr.Status.Provenance.ConfigSource = (*v1beta1.ConfigSource)(meta.RefSource)
-		}
+	if meta != nil && meta.RefSource != nil && pr.Status.Provenance.RefSource == nil {
+		pr.Status.Provenance.RefSource = meta.RefSource
+	}
+	if meta != nil && meta.RefSource != nil && pr.Status.Provenance.ConfigSource == nil {
+		pr.Status.Provenance.ConfigSource = (*v1beta1.ConfigSource)(meta.RefSource)
```

Comment:
I thought this feature was being promoted to beta so it should be guarded behind the alpha feature flag? 🤔

Reply:
This feature was only gated by its dedicated feature flag per https://github.com/tektoncd/pipeline/issues/5632.